### PR TITLE
fix(wallet): fix mercuryo on-ramp

### LINKED ISSUE #20927: Buy through Mercuryo doesn't work
### Description

Attempting to purchase crypto using the mercuryo link we have in the Buy section results in a`Signature is invalid.` error after entering payment details.

The link needs to include a signature based on a secret key and the destination address, which needs to be predetermined when creating the URL parameters.

https://www.figma.com/design/xLs1KYmF4e6WwRTZVJKeUK/Descoped-Wallet?node-id=16428-59853&t=xZ3A8FbB2tBBj86G-0

https://www.figma.com/design/xLs1KYmF4e6WwRTZVJKeUK/Descoped-Wallet?node-id=16428-62308&t=MQSMXdWbuVicDLvV-1

### LINKED ISSUE #21339: Optimism assets not fully fetched in "Popular Assets" section on swap page
### Steps to Reproduce:
1. Go to the swap page and select the Optimism network.
2. Check the "Popular Assets" section in the "Asset to Pay" or "Asset to Receive" sheets.

### Actual Result:
Only 4 tokens are shown in the "Popular Assets" section for Optimism.

https://github.com/user-attachments/assets/1a26a6c4-72b6-4c92-8584-35e459ef14bb

### Expected Result:
More popular tokens should be displayed in the "Popular Assets" section for the Optimism network, as on Desktop

https://github.com/user-attachments/assets/d7f98098-2c2f-44a6-9cc5-2cc5914c88ca

### Devices:
- Pixel 7a, Android 13
- iPhone 11 Pro Max, IOS 17

### Logs:
[logs.zip](https://github.com/user-attachments/files/17161776/logs.zip)

### LINKED ISSUE #21341: Default tokens (SNT, DAI, ETH) are shown with a 0 value in the "Your assets on Mainnet" section
Current issue is with very low priority and can be considered in 2.32 as follow up
 
[issue 5](https://github.com/status-im/status-mobile/pull/21140#issuecomment-2367881461)  is partially fixed. The default tokens are no longer shown as disabled, which is resolved. However, when switching to Network 2, tokens like SNT, DAI, and ETH that exist on Network 1 are shown with a value of 0 on the "Your assets on Mainnet" section, even though they are not present on Network 2
follow up for https://github.com/status-im/status-mobile/pull/21140

### Steps:
1. Create new account or restore an account with snt, dai or eth only on Network 1
2. Select Network 2 for swapping on which tokens are not present, but present on Network 1
3. Go to "Asset to Pay" page

### Actual result:
Default tokens (SNT, DAI, ETH) are displayed with a 0 value in the "Your assets on Mainnet" section, even though they are available on Network 1 but not on Network 2, while Network 2 is selected.

image

### Expected result:
Tokens should not be shown in "Your assets on Mainnet" section if they are not available on the currently selected network. 
https://figma.com/design/AD2JSKg0I8dZcylyiGa62O/Swap-for-Mobile?node-id=76-130105&node-type=frame&t=mccNGM33NrJBogCP-0

### PR DESCRIPTION
**Pull request by briansztamfater**
fixes #20927 
fixes #21339
fixes #21341

### Summary

This PR implements the flow for adding custom parameters on on-ramp providers, which fixes Mercuryo on-ramp links. Also fixes a bug that makes some popular assets not being displayed in the asset selection list.

https://github.com/user-attachments/assets/cf116eaa-8ead-45a4-a12a-5682844ad095

#### Platforms

- Android
- iOS

##### Functional

- wallet / transactions

#### Issue 1
- Open Status
- Log in
- Go to Buy crypto flow
- Select Mercuryo
- Select a network
- Select an asset
- Buy on Mercuryo
- Verify flow is working as expected and that referral is working as expected

#### Issue 2
- Open Status
- Log in
- Go to Swaps
- Select Optimism token
- Select asset to receive
- Verify all the tokens are displayed in the list

status: ready

### CODE REVIEW

**Comment by VolodLytvynenko** [comment]
Hi @briansztamfater, thank you for the PR. No issues from my side, and the PR is ready to be merged. However, I took this PR too early, as it's still under review. If any additional commits are added after the review, let me know and I will retest. Otherwise, the PR can be merged

**Comment by VolodLytvynenko** [comment]
@briansztamfater apologies, I was too quick to confirm the merging of the current PR. I need to clarify a few questions with the design team regarding the current feature

**Comment by shivekkhurana** [comment]
QA happened before Review 😅

**Comment by VolodLytvynenko** [comment]
@briansztamfater Take a look at the found issue

## PR_ISSUE 1: Network drawer is shown when "Mercuryo" is selected on Swap/Send/Bridge pages

### Actual Result:
The drawer with network options is shown.

https://github.com/user-attachments/assets/0829d28d-abdb-402e-9adb-3702203e5937

### Expected result:
The drawer with networks is not shown. User is directly redirected to \select asset to buy' drawer
https://www.figma.com/design/xLs1KYmF4e6WwRTZVJKeUK/Wallet?node-id=16449-14285&node-type=frame&t=V07pw3fkyKsDCJJJ-0

### OS: 
IOS, Android
### Devices:
- Pixel 7a, Android 13
- iPhone 11 Pro Max, IOS 17

**Comment by briansztamfater** [comment]
@VolodLytvynenko thanks for testing, issue 1 should be fixed now!

**Comment by VolodLytvynenko** [comment]
hi @briansztamfater, thank you for the fix. One very low-priority issue is found that can be considered as follow up.  

## ISSUE 2: ETH and DAI are displayed in the "Your assets" section of the swap page, even when the user has a balance of 0 for these assets.

### Steps:
1. Go to the swap page.
2. Open the 'Assets to Pay' or 'Assets to Receive' modals.

### Actual Result:
ETH and DAI are shown in the "Your assets" section, despite the user having a 0 balance for these tokens.

https://github.com/user-attachments/assets/1f3c0fe5-9a36-4318-976a-1d61a7e18e40


### Expected Result:
 "Your assets" section should no contains the asset if the user has no balance for these assets. If user has no assets at all, then the  "Your assets" section should hidden

**Comment by VolodLytvynenko** [comment]
One question: what does mean "one time" and "recurrent tabs"?
image

**Comment by briansztamfater** [comment]
@VolodLytvynenko IMO only in Select Asset to Pay screen makes sense to hide tokens with balance 0. I updated that, let me know if that is okay, we can create a follow up if this needs more thinking.

About one-time and recurrent, some providers allow to buy recurrenty (monthly buys i.e.), so we list them there.

**Comment by briansztamfater** [comment]
> @VolodLytvynenko IMO only in Select Asset to Pay screen makes sense to hide tokens with balance 0. I updated that, let me know if that is okay, we can create a follow up if this needs more thinking.

Should that fix https://github.com/status-im/status-mobile/issues/21341 @VolodLytvynenko ?

**Comment by VolodLytvynenko** [comment]
hi @briansztamfater thank you for fixes. No issues from my side. If any additional commits are added after the review, let me know and I will retest. Otherwise, the PR can be merged. 

[Issue 2](https://github.com/status-im/status-mobile/pull/20962#issuecomment-2397101809) indeed can be considered as a feature

File: src/status_im/contexts/wallet/sheets/buy_token/view.cljs [hunk 1]
```diff
@@ -7,11 +7,20 @@
...
 (defn- crypto-on-ramp-item
-  [{:keys [name description fees logo-url site-url recurrent-site-url]} _ _ {:keys [tab]}]
-  (let [open-url (rn/use-callback (fn []
-                                    (rf/dispatch [:open-url
-                                                  (if (= tab :recurrent) recurrent-site-url site-url)]))
-                                  [site-url recurrent-site-url tab])]
+  [{:keys [name description fees logo-url site-url recurrent-site-url urls-need-parameters]
+    :as   provider}
+   _
+   _
```
Comment (shivekkhurana) [inline src/status_im/contexts/wallet/sheets/buy_token/view.cljs:13]: Leftovers ?

**Review by shivekkhurana — APPROVED** [review]
This PR looks good to go. 

We should merge this after QA, but as discussed on call, we might have to hide Mercuryo due to UK laws.

Regardless, we should have this code on our master branch, so we can enable it whenever we have a decision.

**Comment by VolodLytvynenko** [comment]
Hi @shivekkhurana I didn't test the token purchase process using Mercuryo in this PR; I only checked if the account address and tokens are fetched correctly from Status app, as shown in the screenshot
image

Should we test the entire buy transaction process within such Mercuryo-related PRs, or is it enough to check it during the release phase? (Minimum price to buy tokens within Mercuryo = 25 USD)

**Pull request merged**